minio/rdma.py

```
@@ -120,29 +121,51 @@ def _last_error(lib: ctypes.CDLL) -> str:
     """
     if isinstance(buf, int):
-        return buf, 0
-    if isinstance(buf, (bytes, bytearray)):
-        return ctypes.cast(
-            (ctypes.c_char * len(buf)).from_buffer_copy(buf)
-            if isinstance(buf, bytes)
-            else (ctypes.c_char * len(buf)).from_buffer(buf),
-            ctypes.c_void_p,
-        ).value or 0, len(buf)
+        if buf == 0:
+            raise ValueError("raw RDMA pointer must be non-zero")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)